### PR TITLE
Iterative lookup prototyping

### PR DESCRIPTION
**Pull request by khibino**
Prototype implementation of iterative lookup.

- [x] simple iterative lookup
- [x] resolve CNAME in lookup result
- [x] resolve PTR. process for no glue records case
- [x] NS address reverse lookoup to verify
- [x] balancing multiple NS to send queries.

### CODE REVIEW

**Comment by khibino** [comment]
反復検索のプロトタイプ実装をやってみました @kazu-yamamoto

**Comment by kazu-yamamoto** [comment]
@khibino Would you give me the write access so that I can be a reviwer.

**Comment by kazu-yamamoto** [comment]
`hlint` suggests you many things.

File: src/DNSC/Iterative.hs [hunk 1]
```diff
@@ -0,0 +1,346 @@
+module DNSC.Iterative where
```
Comment (kazu-yamamoto) [inline src/DNSC/Iterative.hs:1]: You should explicitly export functions here.

**Comment by khibino** [comment]
- [ ] hlint
- [ ] explicitly export
- [ ] add test cases
- [ ] balancing destination NS host

File: src/DNSC/Iterative.hs (hunk 1 above)
Comment (khibino) [reply src/DNSC/Iterative.hs]: https://github.com/khibino/dns-resolver/pull/1/commits/579dc30c4899e9c99ff23757560b6d36e8f0444e

**Comment by kazu-yamamoto** [comment]
The test does not pass:

```
query
  rootNS [✔]
  iterative [✘]
  iterative - long [✔]
  query1 - ns [✔]
  query1 - a [✘]
  query1 - aaaa [✘]
  query1 - mx [✘]
  query1 - cname [✘]
  query - a [✘]
```

Would you suggest me how to debug?

**Comment by khibino** [comment]
What is the result of

```
runIterative rootNS "iij.ad.jp."
```

**Comment by kazu-yamamoto** [comment]
```
> runIterative rootNS "iij.ad.jp."
Left (NetworkFailure udp@[2a01:8840:1bc::25]:53: does not exist (No route to host))
```

**Comment by khibino** [comment]
Failure to communicate over IPv6?

**Comment by khibino** [comment]
- [ ] add flag not to send queries to IPv6 address.

**Comment by khibino** [comment]
How about testing new revision with the environment variable `DISABLE_V6_NS=1` ?

**Comment by kazu-yamamoto** [comment]
Bingo! I forgot that my home does not have IPv6.

**Review by kazu-yamamoto — APPROVED** [review]
Now LGTM.
Please merge this PR by yourself.

**Comment by khibino** [comment]
thanks!

**Pull request merged**